### PR TITLE
Add debug info for retry call

### PR DESCRIPTION
**Pull request by vmanilo**
## Changes
- added debug logs

### CODE REVIEW

File: twingate/internal/client/client.go [hunk 1]
```diff
@@ -137,7 +137,19 @@ func customRetryPolicy(ctx context.Context, resp *http.Response, err error) (boo
...
+	}
+
+	errMsg := "<nil>"
+	if err != nil {
+		errMsg = err.Error()
+	}
+
+	log.Printf("[WARN] Failed to call %s, status %s, error: %s", resp.Request.URL.String(), resp.Status, errMsg)
+
+	return true, nil
```
Comment (vmanilo) [inline twingate/internal/client/client.go:152]: original code was not propagated error when should be retried

```
// don't propagate other errors
shouldRetry, _ := baseRetryPolicy(resp, err)
return shouldRetry, nil
```

**Review by twingate-blee — APPROVED** [review]
lgtm

**Review by ekampf — APPROVED** [review]

File: twingate/internal/client/client.go [hunk 2]
```diff
@@ -31,6 +31,8 @@ const (
 	headerAgent         = "User-Agent"
 	headerCorrelationID = "X-Correlation-Id"
 
+	ctxReqIDKey = "Ctx_req_id_key"
```
Comment (ekampf) [inline twingate/internal/client/client.go:34]: lets give this a decent name - `X-Twingate-Request-Id`

Comment (vmanilo) [reply twingate/internal/client/client.go]: yes, updated it

**Review by ekampf — APPROVED** [review]

**Pull request merged**